neuprint/queries.py

```diff
@@ -341,19 +341,14 @@ def fetch_neurons(criteria, *, client=None):
     # return properties individually to avoid a large JSON payload.
     # (Returning a map on every row is ~2x more costly than returning a table of rows/columns.)
     props = compile_columns(client, core_columns=CORE_NEURON_COLS)
-    props.remove('somaLocation')
     return_exprs = ',\n'.join(f'n.{prop} as {prop}' for prop in props)
     return_exprs = indent(return_exprs, ' '*15)[15:]
 
     q = f"""\
         {criteria.global_with(prefix=8)}
         MATCH (n :{criteria.label})
         {criteria.all_conditions(prefix=8)}
-        RETURN {return_exprs},
-               CASE
-                 WHEN n.somaLocation IS NULL
-                 THEN NULL ELSE [n.somaLocation.x, n.somaLocation.y, n.somaLocation.z]
-               END as somaLocation
```

Comment:
Not a criticism, just a comment:

Without this logic in the cypher, the full JSON is returned for the `somaLocation` field:

```
{'coordinates': [31334, 38023, 62575],
 'crs': {'name': 'cartesian-3d',
  'properties': {'href': 'http://spatialreference.org/ref/sr-org/9157/ogcwkt/',
   'type': 'ogcwkt'},
  'srid': 9157,
  'type': 'link'},
 'type': 'Point'}
```

Obviously you know that, since you handle it in `_process_neuron_df()`, below.

I was worried that all that extra data would slow down our queries.  For instance, there is a 3x difference between the speed of the following two queries:


```python
In [13]: %%time
    ...: df = c.fetch_custom("""\
    ...:     MATCH (n:Neuron)
    ...:     WHERE n.somaLocation IS NOT NULL
    ...:     RETURN
    ...:         CASE WHEN n.somaLocation IS NULL
    ...:           THEN NULL
    ...:           ELSE [n.somaLocation.x, n.somaLocation.y, n.somaLocation.z]
    ...:         END as somaLocation
    ...: """)
    ...:
    ...:
CPU times: user 106 ms, sys: 7.78 ms, total: 113 ms
Wall time: 346 ms
```

```python
In [15]: %%time
    ...: df = c.fetch_custom("""\
    ...:     MATCH (n:Neuron)
    ...:     WHERE n.somaLocation IS NOT NULL
    ...:     RETURN n.somaLocation
    ...: """)
    ...:
    ...:
CPU times: user 128 ms, sys: 9.44 ms, total: 137 ms
Wall time: 1.05 s
```

But admittedly, in the real scenario, the extra text will be a smaller percentage of the whole response.  Plus, there's a limit on how many neurons even have coordinate fields.  I guess your simplification is worth the slight penalty.